File: CS/RealTimeSource/Form1.cs
using DevExpress.Data;
using DevExpress.XtraEditors.Repository;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace RealTimeSourceWinForms
{
    public partial class Form1 : Form
    {
        BindingList<Data> Persons;
        int Count = 50;
        Random Random = new Random();
        public Form1()
        {

            InitializeComponent();
            Persons = new BindingList<Data>();
            for (int i = 0; i < Count; i++)
                Persons.Add(new Data
                {
                    Id = i,
                    Text = "Text" + i,
                    Progress = GetNumber()
                });


            RealTimeSource rts = new RealTimeSource()
                        {
                            DataSource = Persons
                        };
            gridControl1.DataSource = rts;

            Timer timer = new Timer();
            timer.Interval = 10;
            timer.Tick += Tick;
            timer.Start();

            gridView1.Columns["Progress"].ColumnEdit = new RepositoryItemProgressBar();
        }
        private void Tick(object sender, EventArgs e)
        {
            int index = Random.Next(0, Count);
            Persons[index].Id = GetNumber();
            Persons[index].Text = "Text" + GetNumber();
            Persons[index].Progress = GetNumber();
        }
        int GetNumber()
        {
            return Random.Next(0, Count);
        }
    }
    public class Data : INotifyPropertyChanged
    {
        private int _Id;
        public string _Text;
        public double _Progress;

        public int Id
        {
            get
            {
                return _Id;
            }
            set
            {
                _Id = value;
                NotifyPropertyChanged("Id");
            }
        }
        public string Text
        {
            get
            {
                return _Text;
            }
            set
            {
                _Text = value;
                NotifyPropertyChanged("Text");
            }
        }
        public double Progress
        {
            get
            {
                return _Progress;
            }
            set
            {
                _Progress = value;
                NotifyPropertyChanged("Progress");
            }
        }

        public event PropertyChangedEventHandler PropertyChanged;
        void NotifyPropertyChanged(string name)
        {
            if (PropertyChanged != null)
                PropertyChanged(this, new PropertyChangedEventArgs(name));
        }
    }
}

File: VB/RealTimeSource/Form1.vb
Imports DevExpress.Data
Imports DevExpress.XtraEditors.Repository
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace RealTimeSourceWinForms

    Public Partial Class Form1
        Inherits Form

        Private Persons As BindingList(Of Data)

        Private Count As Integer = 50

        Private Random As Random = New Random()

        Public Sub New()
            InitializeComponent()
            Persons = New BindingList(Of Data)()
            For i As Integer = 0 To Count - 1
                Persons.Add(New Data With {.Id = i, .Text = "Text" & i, .Progress = GetNumber()})
            Next

            Dim rts As RealTimeSource = New RealTimeSource() With {.DataSource = Persons}
            gridControl1.DataSource = rts
            Dim timer As Timer = New Timer(components)
            timer.Interval = 10
            AddHandler timer.Tick, AddressOf Tick
            timer.Start()
            Dim item As RepositoryItemProgressBar = New RepositoryItemProgressBar()
            gridControl1.RepositoryItems.Add(item)
            gridView1.Columns("Progress").ColumnEdit = item
        End Sub

        Private Sub Tick(ByVal sender As Object, ByVal e As EventArgs)
            Dim index As Integer = Random.Next(0, Count)
            Persons(index).Id = GetNumber()
            Persons(index).Text = "Text" & GetNumber()
            Persons(index).Progress = GetNumber()
        End Sub

        Private Function GetNumber() As Integer
            Return Random.Next(0, Count)
        End Function
    End Class

    Public Class Data
        Implements INotifyPropertyChanged

        Private _Id As Integer

        Public _Text As String

        Public _Progress As Double

        Public Property Id As Integer
            Get
                Return _Id
            End Get

            Set(ByVal value As Integer)
                _Id = value
                NotifyPropertyChanged("Id")
            End Set
        End Property

        Public Property Text As String
            Get
                Return _Text
            End Get

            Set(ByVal value As String)
                _Text = value
                NotifyPropertyChanged("Text")
            End Set
        End Property

        Public Property Progress As Double
            Get
                Return _Progress
            End Get

            Set(ByVal value As Double)
                _Progress = value
                NotifyPropertyChanged("Progress")
            End Set
        End Property

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        Private Sub NotifyPropertyChanged(ByVal name As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(name))
        End Sub
    End Class
End Namespace
